### PR TITLE
Meson improvements

### PR DESCRIPTION
**Pull request by Ericson2314**
See each commit for details. I took this from my old branches.

I think this is strictly better except for one issue: by default (with `stdenv.mkDerivation and Meson) we enable all optional features, and thus enable the s3 store. But the AWS pkg-config file is very bad in specifying `-fno-exceptions` and `std=c++11`, and this breaks the build.

That means the build with AWS was always failing, but now with AWS is the effective default and thus the CI build will fail.

CC @Pamplemousse

N.B. Perhaps we should reframe from rebasing the PR while multiple people are working on it, and we are not sure if/when @edolstra is interested, as it was somewhat difficult to dredge up these changes.

### CODE REVIEW

**Review by p01arst0rm — COMMENTED** [review]
personally, i think meson looks a lot neater when you dont newline the closers :')

**Review by p01arst0rm — COMMENTED** [review]
im pretty sure files were added like this for a reason; albeit i cant remember that reason now! i think it was something to do with a bug in referencing file objects in different directories

**Review by p01arst0rm — COMMENTED** [review]
this file looks excessively messy. Could you give it a do-over maybe? improve formatting n such

File: dependencies/aws-sdk-cpp/meson.build [hunk 1]
```diff
@@ -4,27 +4,43 @@
```
Comment (p01arst0rm) [inline dependencies/aws-sdk-cpp/meson.build:4]: this file looks excessively messy. Could you give it a do-over maybe? improve formatting n such

**Comment by p01arst0rm** [comment]
@Ericson2314 there is mixed usage of `with_[option]` and `enable_[option]`, could you standardise these all to the same syntax?

**Comment by Pamplemousse** [comment]
> personally, i think meson looks a lot neater when you dont newline the closers :')

I strongly disagree:
* it makes it harder to read (quickly scan with eyes where a list/dict ends)
* it makes VCS logs confusing whenever something is added / removed at the end of the list (more line updates than necessary)

**Comment by Pamplemousse** [comment]
LGTM: This seems to be a good improvement of the original PR :+1:

**Comment by p01arst0rm** [comment]
not yet, still need to standardise the with/enable flags

**Comment by p01arst0rm** [comment]
guess i could just do that myself :)

**Comment by Ericson2314** [comment]
Thanks!

**Comment by Ericson2314** [comment]
Thanks!!

**Pull request merged**